src/components/ApplicationDetails/WorkspaceSwitcher.tsx

```diff
@@ -1,35 +1,97 @@
 import React from 'react';
-import { useLocation, useNavigate } from 'react-router-dom';
+import { useLocation, useNavigate, Link } from 'react-router-dom';
+import { Button, Flex, FlexItem } from '@patternfly/react-core';
 import { WorkspaceContext } from '../../utils/workspace-context-utils';
 import { ContextMenuItem, ContextSwitcher } from '../ContextSwitcher';
+import { ItemVisibility } from '../ContextSwitcher/context-switcher-utils';
+import {
+  ContextTab,
+  MenuTabs,
+  WORKSPACE_SWITCHER_STORAGE_KEY,
+} from '../ContextSwitcher/ContextSwitcher';
 
 export const WorkspaceSwitcher: React.FC<
   React.PropsWithChildren<{ selectedWorkspace?: string }>
 > = () => {
   const navigate = useNavigate();
   const { pathname } = useLocation();
-  const { workspace, workspaces } = React.useContext(WorkspaceContext);
+  const { workspace, konfluxWorkspaces = [] } = React.useContext(WorkspaceContext);
 
   const menuItems = React.useMemo(
-    () => workspaces?.map((app) => ({ key: app.metadata.name, name: app.metadata.name })) || [],
-    [workspaces],
+    () =>
+      (Array.isArray(konfluxWorkspaces) &&
```

Comment:
**[Suggestion, not necessary]**: One potential alternate implementation for this could be:
```typescript
const mapWorkspaceToMenuItem = ({ status, metadata, spec }) => ({
  key: status?.space?.name ?? metadata?.namespace,
  name: status?.space?.name ?? metadata?.namespace,
  visibility: spec?.visibility,
});

const menuItems = React.useMemo(() => 
  konfluxWorkspaces?.map(mapWorkspaceToMenuItem) || [],
  [konfluxWorkspaces]
);
```
This simplifies the check for array existence (` konfluxWorkspaces?.map`) and isolates the mapping logic so its easier to test in case the code coverage check is not happy